nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-resources/src/main/resources/bin/nifi.sh

```diff
@@ -245,6 +245,11 @@ SERVICEDESCRIPTOR
         ln -s "/etc/init.d/${SVC_NAME}" "/etc/rc2.d/K65${SVC_NAME}" || { echo "Could not create link /etc/rc2.d/K65${SVC_NAME}"; exit 1; }
         echo "Service ${SVC_NAME} installed"
     fi
+
+    # systemd: generate nifi.service from init.d
+    if [ "$(pidof systemd)" ]; then
```

Comment:
```suggestion
    if [ -d "/run/systemd/system/" ] || [ ! -z "$(pidof systemd 2>/dev/null)" ]; then
```

Reply:
As one additional alternative, since the conditional runs the `systemctl` command, what about checking for the command itself?
```suggestion
    if [ "$(which systemctl)" ]; then
```

Reply:
So, there's this unlikely/weird scenario where some of the systemd components could be installed but systemd isn't actually the init. Should we care about that? I dunno.

Reply:
That's a good question.  It would be helpful to know how likely that scenario is to occur, but if systemd is not being used for initialization, then running `systemctl daemon-reload` shouldn't have an impact on the initialization configuration.  Ultimately, it would be better to handle service installation for systemd using an actual service configuration as opposed to relying on backward compatibility, but as it stands, this update seems like a good step forward.

Reply:
I believe it's important to check if systemd is currently running.
I'd opt for changes suggested by @jfrazee, even though @exceptionfactory option could already cover most of the cases.

Reply:
Can I apply those changes and update the PR?

Reply:
Thanks for the feedback @rmarturano!  The link from @jfrazee in an earlier comment is a helpful reference point for his recommendation, so I will defer to the approach that both of you think is best under the circumstances.